### ISSUE TITLE
[Question] Snap sheet to points

### ISSUE THREAD

**Issue by Jomik** [software-mansion/react-native-reanimated#717]
I am trying to animate something similar to a bottom sheet.
It has to snap to two points. I would like to animate it so that it moves nicely to the snap point when the user releases it.
Something like Timing with Easing, but the distance will vary a lot, so the duration should also vary a lot.
How would I go about that? Should I simply calculate the duration, or is there a better way?

I tried doing a Spring animation, but I would need overshootClamping to work then (#594) and bouncing may not be so nice in this case. 

<details><summary>Code</summary>

```ts
const interaction = (
  translation: Value<number>,
  velocity: Value<number>,
  gestureState: Value<State>,
  min: number,
  max: number
): Animated.Node<number> => {
  const offset = new Value<number>(0);
  const clock = new Clock();

  const position = diffClamp(
    cond(
      eq(gestureState, State.ACTIVE),
      add(offset, translation),
      set(offset, add(offset, translation))
    ),
    min,
    max
  );
  const target = snapPoint(position, velocity, [min, max]);

  const state = {
    finished: new Value<0 | 1>(0),
    velocity: new Value<number>(0),
    position: new Value<number>(0),
    time: new Value<number>(0),
  };
  const config = {
    toValue: new Value<number>(0),
    damping: 15,
    mass: 1,
    stiffness: 150,
    overshootClamping: true,
    restSpeedThreshold: 1,
    restDisplacementThreshold: 1,
    useNativeDriver: true,
  };

  return block([
    cond(
      eq(gestureState, State.ACTIVE),
      [set(state.position, position), stopClock(clock)],
      [
        cond(not(clockRunning(clock)), [
          set(config.toValue, target),
          set(state.velocity, velocity),
          set(state.finished, 0),
          set(state.time, 0),
          startClock(clock),
        ]),
        spring(clock, state, config),
        cond(state.finished, stopClock(clock)),
      ]
    ),
    state.position,
  ]);
};
```

</details>

**Comment by Jomik** [comment]
I have tried calculating the duration for a timing animation, where `min` and `max` are the two snappoints.
```ts
  const target = snapPoint(position, velocity, [min, max]);
  const interval = sub(max, min);
  const distance = abs(sub(target, position));
  const duration = multiply(divide(distance, interval), 400);
```
But it seems to be incorrect. The animation is _super_ fast at weird points. I am sort of unable to make the connection.
I have tried passing the duration AnimatedNode to the config directly, but also tried setting it in the block that resets the state.
